docs/security.rst

```diff
@@ -2,13 +2,29 @@
 
 Security Overview
 -----------------
-The Schema Registry currently supports two security features: communication with a secure Kafka cluster over SSL; and API calls over HTTPS. At this time, ZooKeeper security and Kafka SASL authentication are not yet supported.
+The Schema Registry currently supports all Kafka security features, including: communication and authentication with a
+secure Kafka cluster over SSL; authentication with ZooKeeper over SASL; authentication with Kafka over SASL; and
+end-user REST API calls over HTTPS.
 
 For more details, check the :ref:`configuration options<schemaregistry_config>`.
 
 Kafka Store
 ~~~~~~~~~~~
-The Schema Registry uses Kafka to persist schemas. The following Kafka security configurations are currently supported:
+The Schema Registry uses Kafka to persist schemas, and all Kafka security features are supported by the Schema Registry.
 
-* SSL encryption
-* SSL authentication
+ZooKeeper
+~~~~~~~~~
+ZooKeeper SASL authentication is supported.
+
+Setting up ZooKeeper SASL authentication for the Schema Registry is similar to Kafka's setup. Namely,
+create a keytab for the Schema Registry, create a JAAS configuration file, and set the appropriate JAAS Java properties.
+
+In addition to the keytab and JAAS setup, be aware of the `zookeeper.set.acl` setting. This setting, when set to `true`
+, enables ZooKeeper ACLs, which limits access to znodes.
+
+Important: if `zookeeper.set.acl` is set to `true`, the Schema Registry's service name must be the same as Kafka's, which
+is most likely `kafka`. The Schema Registry's principal must have the correct service name, too. Otherwise, the Schema
+Registry will fail to create the `_schemas` topic, which will cause a leader not available error.
```

Comment:
I agree that having a good error message here is important. However, it may be tough. Let's take the following example: Kafka is not configured to set ACLs in ZooKeeper, but the Schema Registry is. And the Schema Registry's principal is not Kafka's.

In this scenario, the Schema Registry will create the topic without error. Then, the broker will throw an exception when electing a leader for the topic's partitions, because it can't modify ZooKeeper. That error shows up in the state change log on the broker. Meanwhile, the Schema Registry attempts to consume from the topic, eventually failing with a timeout because the partition doesn't have a leader.

Given this scenario, I'm not sure how to give the user a better error message. Any ideas?